tests/NHapi.NUnit/Parser/PipeParserTests.cs

```diff
@@ -190,5 +190,202 @@ public void ParseWithNullConfigThrows()
+        /// See http://sourceforge.net/p/hl7api/bugs/171/.
+        /// </summary>
+        [Test]
+        public void GreedyMode()
+        {
+            var msg = new Model.V251.Message.OML_O21();
+            SetMessageHeader(msg, "OML", "O21", "T");
+
+            for (var i = 0; i < 5; i++)
+            {
+                var orc = msg.GetORDER(i).ORC;
+                var obr4 = msg.GetORDER(i).OBSERVATION_REQUEST.OBR.UniversalServiceIdentifier;
+
+                orc.OrderControl.Value = "NW";
+                orc.PlacerOrderNumber.EntityIdentifier.Value = "ORCID1";
+                orc.PlacerOrderNumber.NamespaceID.Value = "HCE";
+                orc.PlacerGroupNumber.EntityIdentifier.Value = "grupo";
+
+                obr4.Identifier.Value = "STDIO1";
+                obr4.NameOfCodingSystem.Value = "LOINC";
+            }
+
+            // Parse and encode
+            var pp = new PipeParser();
+            msg = pp.Parse(
+                pp.Encode(msg),
+                new ParserConfiguration { NonGreedyMode = true }) as Model.V251.Message.OML_O21;
+            Assert.NotNull(msg);
+
+            for (var i = 0; i < 5; i++)
+            {
+                var actual = msg.GetORDER(i).ORC.OrderControl.Value;
+                Assert.AreEqual("NW", actual);
+
+                actual = msg.GetORDER(i).OBSERVATION_REQUEST.OBR.UniversalServiceIdentifier.Identifier.Value;
+                Assert.AreEqual("STDIO1", actual);
+            }
+
+            // Now turn off greedy mode
+            msg = pp.Parse(
+                pp.Encode(msg),
+                new ParserConfiguration { NonGreedyMode = false }) as Model.V251.Message.OML_O21;
+            Assert.NotNull(msg);
+
+            {
+                var actual = msg.GetORDER(0).ORC.OrderControl.Value;
+                Assert.AreEqual("NW", actual);
+
+                actual = msg.GetORDER(0).OBSERVATION_REQUEST.OBR.UniversalServiceIdentifier.Identifier.Value;
+                Assert.AreEqual("STDIO1", actual);
+            }
+
+            for (var i = 1; i < 5; i++)
+            {
+                var actual = msg.GetORDER(i).ORC.OrderControl.Value;
+                Assert.IsNull(actual);
+
+                actual = msg.GetORDER(i).OBSERVATION_REQUEST.OBR.UniversalServiceIdentifier.Identifier.Value;
+                Assert.IsNull(actual);
+            }
+        }
+
+        /// <summary>
+        /// This test is ported from HAPI:
+        /// https://github.com/hapifhir/hapi-hl7v2/blob/3333e3aeae60afb7493f6570456e6280c0e16c0b/hapi-test/src/test/java/ca/uhn/hl7v2/parser/NewPipeParserTest.java#L217
+        /// See http://sourceforge.net/p/hl7api/bugs/171/.
+        /// </summary>
+        [Test]
+        public void MoreGreedyMode()
+        {
+            // OML_O21 messages extracted from https://sourceforge.net/p/hl7api/bugs/171/attachment/IHE_LTW_Examples.java
+            var messages = new System.Collections.Generic.List<string>
```

Comment:
We could just read these from a file (if one exists) and splitting the string on some message delimiter?
perhaps with [File.ReadAllTextAsync](https://docs.microsoft.com/en-us/dotnet/api/system.io.file.readalltextasync?view=net-5.0)